### PR TITLE
Point to subctl repo for devel binaries download

### LINKED ISSUE #21: Change URLs to download subctl binary from new subctl repository
Part of [Epic: Move subctl to its own repo](https://github.com/submariner-io/enhancements/issues/89#)

### PR DESCRIPTION
**Pull request by Jaanki**
Closes: #21

Depends on https://github.com/submariner-io/get.submariner.io/pull/29

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
The version-handling check for devel needs to be updated.

File: index.html [hunk 1]
```diff
@@ -69,7 +69,11 @@
            draft_filter="grep \-rc"
            ;;
     latest) url=https://api.github.com/repos/submariner-io/releases/releases/latest ;;
-    devel|release*|feature*)
+    devel)
+        echo "https://github.com/submariner-io/subctl/releases/download/subctl-${VERSION}/subctl-${VERSION}-${os}-${architecture}.tar.xz"
+        return
+        ;;
+    release*|feature*)
```
Comment (tpantelis) [inline index.html]: @skitt and I were actually just discussing this. `devel` can retrieve from `subctl` as you've done here. `release*` and `feature*` need to as well but going forward - prior versions need to remain retrieving from `submariner-operator`. So:

```
feature-multi-active-gw|release-0.10|release-0.11|release-0.12)
    echo "https://github.com/submariner-io/submariner-operator/releases/download/subctl-${VERSION}/subctl-${VERSION}-${os}-${architecture}.tar.xz"
     return
     ;;
devel|release*|feature*)
    echo "https://github.com/submariner-io/subctl/releases/download/subctl-${VERSION}/subctl-${VERSION}-${os}-${architecture}.tar.xz"
     return
     ;;

```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by tpantelis** [comment]
This [job](https://github.com/submariner-io/get.submariner.io/runs/6389436940?check_suite_focus=true) fails b/c this check:

`[[ "$(~/.local/bin/subctl version)" =~ ^"subctl version: subctl-devel"(-[0-9]+-g[a-f0-9]+)?$ ]]`

is incorrect - it's been failing for quite a while. It seems we used to name the version "_subctl-devel_" but it's now just "_devel_". Please make this change as well.

File: index.html [hunk 2]
```diff
@@ -70,6 +70,10 @@
            ;;
     latest) url=https://api.github.com/repos/submariner-io/releases/releases/latest ;;
     devel|release*|feature*)
+        echo "https://github.com/submariner-io/subctl/releases/download/subctl-${VERSION}/subctl-${VERSION}-${os}-${architecture}.tar.xz"
+        return
+        ;;
+    feature-multi-active-gw|release-0.10|release-0.11|release-0.12)
```
Comment (tpantelis) [inline index.html]: This needs to be above L72 otherwise, eg, `release-0.12` will match L72 and fail.

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by tpantelis** [comment]
The [job](https://github.com/submariner-io/get.submariner.io/runs/6391421364?check_suite_focus=true) for `release-0.8` also failed. Perhaps we should change it to check for `release-0.12` but, either way, should we handle `release-0.8` (@skitt )? I guess it wouldn't hurt to include `release-0.8` to `release-0.12`.

**Comment by Jaanki** [comment]
> The [job](https://github.com/submariner-io/get.submariner.io/runs/6391421364?check_suite_focus=true) for `release-0.8` also failed. Perhaps we should change it to check for `release-0.12` but, either way, should we handle `release-0.8` (@skitt )? I guess it wouldn't hurt to include `release-0.8` to `release-0.12`.

I was thinking the same. From 8-12, get from operator repo and after that from subctl repo.

File: index.html (hunk 2 above)
Comment (skitt) [inline index.html]:
```suggestion
    feature-multi-active-gw|release-0.?|release-0.1[012])
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: index.html (hunk 2 above)
Comment (tpantelis) [reply index.html]: We can keep the `release-0.8` test job and add one for `release-0.12`. Once the `release-0.13` branch is created we can add a test for that.

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**